src/c128time.c

```diff
@@ -7,33 +7,83 @@
 #include <peekpoke.h>
 #include <stdlib.h>
 #include <stdio.h>
-#include "general.h"
+#include <c128.h>
 #include <stdint.h>
 
-char *get_time(void) {
-  uint32_t h = PEEK(0x00A0) * 65536, m = PEEK(0x00A1) * 256, s = PEEK(0x00A2);
-  static char buffer[9];
-  BYTE hrs, min;
+#include "bcd2dec.h"
+#include "general.h"
+#include "globals.h"
+
+void update_time(void) {
+  uint8_t bcd_hour, hour, min, sec, tenth;
+
+  /* Read the hour register first to stop the clock from updating the external
+   * registers from the internal (still ticking!) CIA registers. */
```

Comment:
Here and elsewhere: what does CIA stand for? Can we expand the acronym?

Reply:
The Complex Interface Adapter. Should I maybe link this? 
https://www.c64-wiki.com/wiki/CIA

Reply:
Given we're on a Commodore machine it's pretty obvious, but the link does not hurt either way.

Reply:
Yeah, adding the link would be great :)